### PR TITLE
feat(select): add support select

### PR DESCRIPTION
**Pull request by wKich**
- Added these interactors:
  - `NativeSelect`
  - `NativeMultiSelect`
  - `Select`
  - `MultiSelect`

For now the `Select` and `MultiSelect` interactions support only the [Chip](https://material-ui.com/components/chips/) value render, see example https://material-ui.com/components/selects/#multiple-select

@cowboyd Do we want to support [optgroup](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/optgroup)?

### CODE REVIEW

**Comment by cowboyd** [comment]
> @cowboyd Do we want to support [optgroup](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/optgroup)?

Like having a separate optgroup Interactor?

**Comment by wKich** [comment]
No, I mean to change `Selector` actions API. Because it's possible to have similar option items with the same text, but in different groups.
I suggest this API:
```ts
{
  choose: async (interactor, groupOrValue: string, value?: string) => {
      if (value) await interactor.find(SelectGroupOption(groupOrValue)).find(SelectOption(value)).choose();
      else await interactor.find(SelectOption(groupOrValue)).choose();
  },
}
```

**Comment by cowboyd** [comment]
> No, I mean to change `Selector` actions API. Because it's possible to have similar option items with the same text, but in different groups.

Yes, I see. This makes sense. Logically, the overrides in typescript would be something like:

```ts
function choose(value: string): Interaction;
function choose(value: string, withinOptionGroup: string): Interaction;
```

It sounds like this is something that should be added to core. However, I do think that we should expose the  optgroup interactor because people may want to assert on the `enabled/disabled` state:

```js
// if you set your country as Australia
await Select("Country").choose("Australia");

// then you shouldn't be able to select and dentists in South America
await Select("Dentists").group("South America").is({ disabled: true });
```

But that asks the question then if we have that API, why not just force you to go through the option group if you need to disambiguate:

```js
//option 1
await Select("Dentists").group("Australia").choose("Bob Smith");
// option 2
await Select("Dentists).choose("Bob Smith", { group: "Australia" });
// option 3
await Select("Dentists).choose("Bob Smith", "Australia" });
```

It seems to me like option 1 is mandatory, and we can maybe add options 2 or 3 as shortcuts?

File: src/select.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  .filters({
+    values: (element) =>
+      Array.from(element.querySelectorAll(".MuiChip-root > .MuiChip-label") ?? [])
+        .map((chip) => (isHTMLElement(chip) ? chip.innerText : null))
+        .filter(isDefined),
+  })
+  .actions({
+    choose: async (interactor, value: string) => {
+      try {
+        await interactor.has({ values: [value] });
```
Comment (cowboyd) [inline src/select.ts]: We don't want to wait for the entire timeout period to proceed if the selection is already there that we need. Let's extract the values filter into a stand-alone function that we can invoke from `interactor.peform`.

**Review by cowboyd — APPROVED** [review]

**Pull request merged**